### PR TITLE
fix yaml & use entry_points

### PR DESCRIPTION
**Pull request by sp3c73r2038**
- fix `yaml.load` with `yaml.safe_load`
- use `entry_points` instead of legacy `scripts` in `setup.py`
  as installing using pip from git scm will not update `bin/pipelines` script
- leave `bin/pipelines` for back-compatibility purpose
- fixed filter matching bug in ghauth handler

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -38,7 +38,11 @@
         'pipelines.api',
         'pipelines.plugins',
         'pipelines.plugin'],
-    scripts=['bin/pipelines'],
+    entry_points={
+        'console_script': [
```
Comment (pallxk) [inline setup.py:42]: console_script**s**?

Comment (sp3c73r2038) [reply setup.py]: typo, fixed

**Comment by sp3c73r2038** [comment]
filter matching bug in ghauth handler is now fixed

File: pipelines/api/ghauth.py [hunk 2]
```diff
@@ -184,16 +184,26 @@ def cb(*args):
...
             teams = r.json()
 
-            for org, team in allowed_teams:
-                if filter(lambda x: x['slug'].lower() == team.lower() and x.get('organization',{}).get('login', '').lower() == org.lower(), teams):
-                    log.debug('Allowed access to github user for team %s/%s' % (org, team))
+            user_teams_set = set()
+            for x in teams:
+                o = x.get('organization',{}).get('login', '').lower()
+                t = x.get('slug', '').lower()
+                user_teams_set.add((o, t))
```
Comment (kaleocheng) [inline pipelines/api/ghauth.py:197]: if `o` or `t` == `''` in here. then no need to add them to user_teams_set I guess

File: pipelines/api/ghauth.py [hunk 3]
```diff
@@ -184,16 +184,26 @@ def cb(*args):
...
+            log.debug('user teams: %s' % user_teams_set)
+
+            intersection = allowed_teams_set.intersection(user_teams_set)
+            log.debug('intersection: %s' % intersection)
+            if len(intersection) > 0:
+                    log.debug('Allowed access to github user for team %s' % (intersection))
                     user['username'] = user.get('login', 'unknown')
```
Comment (kaleocheng) [inline pipelines/api/ghauth.py:204]: not related to this PR but github login username should be `mandatory` filed.

File: pipelines/cli.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+import pprint
+
+import docopt
+from getpass import getpass
+
+from pipelines import __version__, PipelinesError
+from pipelines.api import server
+from pipelines.pipeline.utils import random_secret
+
+MIN_PASSWORD_LEN = 4
```
Comment (kaleocheng) [inline pipelines/cli.py:35]: let's bump it to 12

File: pipelines/api/ghauth.py (hunk 3 above)
Comment (sp3c73r2038) [reply pipelines/api/ghauth.py]: check user['login'] beforehand.

**Pull request merged**